test/utils/update/managing.officer.mapper.spec.ts

```diff
@@ -178,16 +184,48 @@ describe("Test mapping to managing officer", () => {
 
     test('that undefined is returned when both residential and principal addresses are undefined', () => {
       const mockDataWithUndefinedAddresses: ManagingOfficerPrivateData[] =
-      [
-        {
-          ...MOCK_MANAGING_OFFICERS_PRIVATE_DATA[0],
-          residentialAddress: undefined,
-          principalAddress: undefined,
-          hashedId: 'hashedId1',
-        },
-      ];
-      const address = mapMoPrivateAddress(mockDataWithUndefinedAddresses, 'mo-undefined-ch-ref', false);
-      expect(address).toBeUndefined();
+        [
+          {
+            ...MOCK_MANAGING_OFFICERS_PRIVATE_DATA[0],
+            residentialAddress: undefined,
+            principalAddress: undefined,
+            hashedId: 'hashedId1',
+          },
+        ];
+      const address = mapMoPrivateAddress(mockDataWithUndefinedAddresses, 'hashedId1', false);
+      expect(address).toEqual({});
```

Comment:
I had to change this test as I realised it wasn't testing properly (wrong hashed Id). When I changed it I noticed that we don't return _undefined_ from mapBOMOAddress function, that function seems to return an empty object.